ansible/library/ptf_portchannel.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python
+
+DOCUMENTATION = '''
+module:  ptf_portchannel
+
+short_description: manage portchannel interface in PTF container
+description: start/stop portchannel interface in PTF container with certain configurations
+
+Options:
+    - option-name: cmd
+      description: An action string as [start|stop]
+      required: True
+    - option-name: portchannel_config
+      description: A dict to indicate the portchannel configuration. E.G. {"PortChannel101": { "intfs": [0, 4] } }
+      required: True
+'''
+
+
+EXAMPLES = '''
+- name: Start PTF portchannel
+  ptf_portchannel:
+    cmd: "start"
+    portchannel_config: "{{ portchannel_config }}"
+'''
+
+
+import os
+import re
+import traceback
+
+import jinja2
+from ansible.module_utils.basic import *
+
+
+portchannel_conf_path = "/etc/portchannel"
+
+
+portchannel_conf_tmpl = '''\
+{
+    "device": "{{ name }}",
+    "runner": {
+        "active": true,
+        "name": "lacp",
+        "min_ports": 1
+    },
+    "ports": {
+{%- for intf in intfs %}
+        "{{ intf }}": {}{{ "," if not loop.last else "" }}
+{%- endfor %}
+    }
+}
+'''
+
+
+portchannel_supervisord_path = "/etc/supervisor/conf.d"
+
+
+portchannel_supervisord_conf_tmpl = '''\
+[program:portchannel-{{ name }}]
+command=/usr/bin/teamd -r -t {{ name }} -f '''+ portchannel_conf_path + '''/{{ name }}.conf
```

Comment:
>portchannel_conf_path [](start = 48, length = 21)

What is the benefit to create a conf file vs simply use command line arguments?

Reply:
If it's a simple command line instead of a conf file, I have to handle some of complex escape characters to ' and ", meanwhile, I think it's not good for readability because all of them should in one line. If the future.